src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -304,15 +304,16 @@ public synchronized IndexShard createShard(int sShardId) throws ElasticsearchExc
             indicesLifecycle.beforeIndexShardCreated(shardId, indexSettings);
 
             logger.debug("creating shard_id {}", shardId);
-
+            // if we are on a shared FS we only own the shard (ie. we can safely delete it) if we are the primary.
+            final boolean ownsShard = IndexMetaData.usesSharedFilesystem(indexSettings) == false || (primary && IndexMetaData.usesSharedFilesystem(indexSettings));
```

Comment:
I wonder if we should name this after what we use it for ie. `canDeleteShardContent`?